rir/src/compiler/native/types_llvm.cpp

```diff
@@ -316,7 +316,7 @@ int initializeTypes(LLVMContext& context) {
         llvm::FunctionType::get(t::Double, {t::SEXP}, false);
 
     NativeBuiltins::colonInputEffects.llvmSignature =
-        llvm::FunctionType::get(t::Int, {t::SEXP, t::SEXP, t::Int}, false);
+        llvm::FunctionType::get(t::SEXP, {t::SEXP, t::SEXP, t::Int}, false);
```

Comment:
why did you change this? this is super inefficient, as it will end up going through `unboxIntLgl`

Reply:
It was necessary because `ColonInputEffects` uses `NativeType::test` and gets passed to a `Branch` instruction. There was a bug where the native backend was boxing the returned integer into a different value, which was messing up the `Branch`. It's weird too because the native backend does use integers for `NativeType::test`, so the bug doesn't usually occur, it was the one that only occurred in the release build in Docker containers and I couldn't reproduce locally or in PRL5.

Reply:
But test boxing is handled in https://github.com/reactorlabs/rir/blob/master/rir/src/compiler/native/lower_llvm.cpp#L847